no-vary-query.md

```diff
@@ -0,0 +1,264 @@
+
+#### … other browser caches
+
+There are several other URL-keyed caches on the web platform. Examples are:
+
+- The [list of available images](https://html.spec.whatwg.org/#the-list-of-available-images)
+- [Module maps](https://html.spec.whatwg.org/#module-map)
+- [Shared workers](https://html.spec.whatwg.org/#shared-workers-and-the-sharedworker-interface:concept-sharedworkerglobalscope-constructor-url)
+- [Resources preloaded via `<link rel="preload">`](https://html.spec.whatwg.org/#map-of-preloaded-resources)
+
+The proposal does _not_ update cache key computation for these caches. This is because:
+
+- None of them respect the `Vary` header today
+- All of them are keyed by the _request_ URL, and not by the _response_ URL, whereas `No-Vary-Query` is a property provided by the response.
+
+### Interaction with prerendering activation
+
+Prerendering has an extra consideration because it's possible for the response to get used based on the original URL, and then eventually activated at which point we know the final URL. For example, consider the [previous underwater phone article example](#carrying-data-not-yet-determined-at-the-time-of-preloading), assuming that `/articles/new-underwater-phone` responds with `No-Vary-Query: "via"`. If the browser chooses to use the `<script type="speculationrules">` to prerender `/articles/new-underwater-phone`, then that resource will be fetched and prerendered with no query parameters. Later, the user might click on the hero image, at which point we know the "real" URL that should be shown to the user is `/articles/new-underwater-phone?via=heroimage`.
+
+To resolve this, we specify that prerendering [activation](https://wicg.github.io/nav-speculation/prerendering.html#prerendering-browsing-context-activate) involves changing the page's URL before firing the `prerenderingchange` event. This URL change is done using the [URL and history update steps](https://html.spec.whatwg.org/#url-and-history-update-steps), i.e. the same mechanism underlying `history.replaceState()`.
+
+With this setup, prerendered pages which care about a given query parameter, but don't want mismatches to prevent a prerender, can delay processing of the query parameter until activation time. For example:
+
+```js
+document.addEventListener("prerenderingchange", () => {
+  const via = (new URLSearchParams(location.search)).get("via");
+  analytics.send("via", via);
+});
+```
+
+### Interaction with redirects
+
+TODO
+
+### Interaction with storage partitioning
+
+This proposal takes place fully within the bounds of the [client side storage partitioning](https://github.com/privacycg/storage-partitioning/) work. It is only about calculating the cache key within a given partition. It does not affect the computation of the partition key itself.
+
+## Alternatives considered
+
+TODO
```

Comment:
Some of these do exist in the internal outline doc.

Reply:
Yep, just ran out of time yesterday; will be porting them over today.